frontend/src/Frontend/Log.hs

```diff
@@ -0,0 +1,83 @@
+{-# LANGUAGE TemplateHaskell #-}
+module Frontend.Log
+  ( LogCfg (..)
+  , Logger (..)
+  , HasLogCfg (..)
+  , HasLogger (..)
+  , LogLevel (..)
+  , formatLogMessage
+  , defaultLogger
+  , logOn
+  , logPromptly
+  , logStrPromptly
+  , fmtLogTH
+  , logDefTH
+  , defLogDebug
+  ) where
+
+import Control.Lens (view,(.~))
+import Control.Monad.IO.Class (MonadIO, liftIO)
+import Control.Monad.Logger (LogLevel (..), LogStr, liftLoc, defaultLogStr, toLogStr, defaultLoc)
+import System.Log.FastLogger (fromLogStr)
+import System.IO (stdout)
+import Data.Function ((&))
+import Data.ByteString (hPut)
+import Data.Text (Text)
+
+import Language.Haskell.TH.Syntax (Q, Exp, qLocation)
+
+import Reflex
+import Reflex.Network.Extended (Flattenable (..))
+
+import Common.Foundation (makePactLenses)
+
+newtype LogCfg t = LogCfg
+  { _logCfg_logMessage :: Event t (LogLevel, Text)
+  }
+
+makePactLenses ''LogCfg
+
+data Logger t = Logger
+  { _log_formatMessage :: LogLevel -> Text -> LogStr
+  , _log_logPromptly :: LogLevel -> Text -> IO ()
+  , _log_logStr :: LogStr -> IO ()
```

Comment:
I'm thinking this shouldn't exist so that we always force folk to choose the log level so that we could filter out certain log levels later? And then if you don't have it you probably don't need formatMessage.

Reply:
Removed this along with the TH stuff that was a bit too janky.